bika/lims/browser/bika_listing.py

```diff
@@ -246,20 +247,15 @@ def submitTransition(self, action, came_from, items):
                     else:
                         success, message = doActionFor(item, action)
                     if success:
-                        transitioned.append(item.id)
+                        transitioned.append(item.UID())
                     else:
                         self.addPortalMessage(message, 'error')
 
         # automatic label printing
-        if transitioned \
-                and action == 'receive' \
-                and 'receive' in self.portal.bika_setup.getAutoPrintStickers():
-            q = "/sticker?template=%s&items=" % \
-                (self.portal.bika_setup.getAutoStickerTemplate())
-            # selected_items is a list of UIDs (stickers for AR_add use IDs)
-            q += ",".join(transitioned)
-            dest = self.context.absolute_url() + q
-            self.destination_url = dest
+        auto_stickers_action = self.portal.bika_setup.getAutoPrintStickers()
+        if transitioned and action == auto_stickers_action:
+            self.request.form['uids'] = transitioned
+            self.workflow_action_print_stickers()
```

Comment:
Thanks for removing that backslash madness, which always looks to me like a huge regular expression. Finally, I can understand that little piece of code:)

However, looking at the return value `return len(transitioned), dest` I see nowhere this `dest` value set differently than `None`. So I don't understand that logic what happens then. Should `dest` be `self.destination_url`?

We should really get away from using classes as multi-state-machines with implicit usage of instance variables, always gives me headaches...

Reply:
To be honest, did not recall on that "dest" variable. Since the redirection is managed by `workflow_action_print_stickers`, I think it should be totally safe to remove this `dest`, but I will check if anyone is expecting this tuple return (I'd say that no one).

Reply:
In any case, there is no sense to have this whole logic inside `submitTransition`, especially because this workflow logic should not live inside `bika_listing` imo, rather in `workflow.<portal_type>.events.*` . I will take care of this soon.

Reply:
Yes, this logic needs to go out from the listing code, agree.
I think so as well that other places handle that `None` URL gracefully, at least the places I can remember. So I'm fine to let that in, ok?

Reply:
Ok, then I will just do `dest = self.destination_url` (its value is already assigned in `workflow_action_print_stickers`) as a temporary solution until we get rid of this crap.